js_modules/dagster-ui/packages/ui-core/src/ui/autolinking.ts

```diff
@@ -0,0 +1,201 @@
+
+    '|',
+
+    // ip address
+    '(?:[0-9]{1,3}\\.){3}[0-9]{1,3}',
+    ')',
+    ')',
+
+    // :port (optional)
+    '(?::d*)?',
+
+    '|',
+
+    // mailto:username@password.com
+    'mailto:\\/*(?:\\w+\\.|[\\-;:&=\\+\\$.,\\w]+@)[A-Za-z0-9\\.\\-]+',
+    ')',
+
+    // optionally followed by:
+    '(',
+    // URL components
+    // (last character must not be puncation, hence two groups)
+    '(?:[\\+=~%\\/\\.\\w\\-_@:]*[\\+~%\\/\\w\\-:_])?',
+
+    // optionally followed by one or more query string ?asd=asd&as=asd type sections
+    "(?:\\?[\\-\\+=&;:%@$\\(\\)'\\*\\/~\\!\\.,\\w_]*[\\-\\+=&;~%@\\w_\\/])*",
+
+    // optionally followed by a #search-or-hash section
+    "(?:#['\\$\\&\\(\\)\\*\\+,;=\\.\\!\\/\\\\\\w%-?]*[\\/\\\\\\w])?",
+    ')?',
+    ')',
+  ];
+
+  return new RegExp(parts.join(''), 'gi');
+}
+
+function runOnTextNode(node: Node, matchers: AutolinkMatcher[]) {
+  // Important: This method iterates through the matchers to find the LONGEST match,
+  // and then inserts the <a> tag for it and operates on the remaining previous / next
+  // siblings.
+  //
+  // It looks for the longest match so that URLs that contain phone number fragments
+  // are parsed as URLs, etc. Here's an example:
+  // https://www.zoom.com/j/9158385033
+  //
+  // We might be able to "order" the regexps carefully to achieve the same result, but
+  // that would be pretty fragile and this "longest" algo more clearly expresses the
+  // behavior we really want.
+  //
+  if (node.parentElement) {
+    const withinScript = node.parentElement.tagName === 'SCRIPT';
+    const withinStyle = node.parentElement.tagName === 'STYLE';
+    const withinA = node.parentElement.closest('a') !== null;
+    if (withinScript || withinA || withinStyle) {
+      return;
+    }
+  }
+  if (!node.textContent) {
+    return;
+  }
+  const nodeTextContentLen = node.textContent.trim().length;
+  if (nodeTextContentLen < 4 || nodeTextContentLen > MAX_ATTEMPTED_TEXT_CONTENT_LENGTH) {
+    return;
+  }
+
+  let longest: {prefix: string; match: RegExpMatchArray} | null = null;
+  let longestLength = null;
+  for (const {prefix, regexp} of matchers) {
+    regexp.lastIndex = 0;
+    const match = regexp.exec(node.textContent);
+    if (match !== null) {
+      if (!longestLength || match[0].length > longestLength) {
+        longest = {prefix, match};
+        longestLength = match[0].length;
+      }
+    }
+  }
+
+  if (longest) {
+    const {prefix, match} = longest;
+    const href = `${prefix}${match[0]}`;
+    const range = document.createRange();
+    range.setStart(node, match.index || 0);
+    range.setEnd(node, (match.index || 0) + match[0].length);
+    const aTag = wrapRangeInNode(range, 'A') as HTMLAnchorElement;
+    aTag.href = href;
+    aTag.rel = 'nofollow noreferrer';
+    aTag.target = '_blank';
+    aTag.title = href;
+  }
+}
+
+export function autolinkTextContent(el: HTMLElement, options: {useIdleCallback: boolean}) {
```

Comment:
Are there any tests available for this as well?

Reply:
Ahh will see if we can pull in some tests, it does seem like it'd be easy to verify a before + after node tree.